system-test/src/util/docker-compose.js

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs')
+
+const { promisify } = require('util')
+const exec = promisify(require('child_process').exec)
+
+const LOGS_DIRECTORY = 'logs'
+
+function DockerCompose (composeFiles, envFile) {
+  if (!Array.isArray(composeFiles)) {
+    composeFiles = [composeFiles]
+  }
+  const composeFilesString = composeFiles.map(file => `-f ${file}`).join(' ')
+  return async command => await exec(`docker-compose ${composeFilesString} --env-file ${envFile} ${command}`)
+}
+
+async function writeDockerLogs (dockerComposeFn, services) {
+  const escapedTestName = expect.getState().currentTestName.split(/,| |:/).join('_')
+  const logFile = `${LOGS_DIRECTORY}/${escapedTestName}.log`
+  if (!fs.existsSync(LOGS_DIRECTORY)) {
+    fs.mkdirSync(LOGS_DIRECTORY)
+  }
+  const fd = fs.openSync(logFile, 'w')
+  for (const service of services) {
+    const logs = await dockerComposeFn(`logs --no-color ${service}`)
+    fs.writeSync(fd, logs.stdout)
+  }
```

Comment:
Very minor nitpick, and very likely does not matter at all performance wise, but there must be a way to just pipe that through without saving the whole log in a variable, right? :smile:

Reply:
It should be possible in general but `docker-compose logs` already returns a string and not a stream, so I don't think it makes any difference here.